[Bug]: prettier is now a mandatory dependency of core (since 10.1.0)

### Describe the bug

With the 10.1.0 update, `prettier` apparently became a mandatory dependency of `core` although it is [still listed as an optional peer dependency.](https://github.com/storybookjs/storybook/blob/next/code/core/package.json#L339-L346)

I would assume that this is a regression and it should not actually have become a mandatory dependency.

When building our storybook, this leads to:

```
node_modules/storybook/dist/common/index.d.ts:8:27 - error TS2307: Cannot find
│  module 'prettier' or its corresponding type declarations.

│  8 import * as prettier from 'prettier';
```

Because `node_modules/storybook/dist/common/index.d.ts` has line 8:
```
import * as prettier from 'prettier';
```
whereas this line was missing in earlier versions:

- Storybook 10.0.8 did not have this issue.
- Storybook 10.1.0 does have this issue.
- Storybook 10.1.2 does have this issue.


### Reproduction link

sorry, none. I understand the value of reproduction repos, but in this case it is probably related to a build tooling change, so any project w/o prettier installed should do

### Reproduction steps

_No response_

### System

```bash
│  Storybook Environment Info:
│

│  System:
│  OS: Linux 5.15 Linux Mint 21.3 (Virginia)
│  CPU: (12) x64 AMD Ryzen 5 5600 6-Core Processor
│  Shell: 5.1.16 - /bin/bash
│  Binaries:
│  Node: 24.11.1 - /home/jan/.nvm/versions/node/v24.11.1/bin/node
│  npm: 11.6.4 - /home/jan/.nvm/versions/node/v24.11.1/bin/npm <----- active
│  Browsers:
│  Chrome: 142.0.7444.175
│  Firefox: 144.0
│  npmPackages:
│  @storybook/addon-docs: ^10.0.7 => 10.1.2
│  @storybook/addon-links: ^10.0.7 => 10.1.2
│  @storybook/angular: ^10.0.7 => 10.1.2
│  eslint-plugin-storybook: ^10.0.7 => 10.1.2
│  storybook: ^10.1.2 => 10.1.2
```

### Additional context

_No response_

--- a/code/core/src/common/utils/formatter.ts
+++ b/code/core/src/common/utils/formatter.ts
@@ -1,8 +1,46 @@
-export async function getPrettier() {
-  return import('prettier').catch((e) => ({
-    resolveConfig: async () => null,
-    format: (content: string) => content,
-  }));
+// Prettier interface definition
+// Note: We want to avoid importing prettier directly to prevent bundling its type import
+// because prettier is an optional peer dependency and might not be available
+interface Prettier {
+  resolveConfig: (filePath: string, options?: { editorconfig?: boolean }) => Promise<any>;
+  format: (content: string, options?: any) => Promise<string> | string;
+  check: (content: string, options?: any) => Promise<boolean>;
+  clearConfigCache: () => Promise<void>;
+  formatWithCursor: (
+    content: string,
+    options?: any
+  ) => Promise<{ formatted: string; cursorOffset: number }>;
+  getFileInfo: (
+    filePath: string,
+    options?: any
+  ) => Promise<{ ignored: boolean; inferredParser: string | null }>;
+  getSupportInfo: () => Promise<{ languages: any[]; options: any[] }>;
+  resolveConfigFile: (filePath?: string) => Promise<string | null>;
+  version: string;
+  AstPath: any;
+  doc: any;
+  util: any;
+}
+
+export async function getPrettier(): Promise<Prettier> {
+  try {
+    return await import('prettier');
+  } catch {
+    return {
+      AstPath: class {} as any,
+      doc: {} as any,
+      util: {} as any,
+      version: '0.0.0-fallback',
+      resolveConfig: async () => null,
+      format: (content: string) => Promise.resolve(content),
+      check: () => Promise.resolve(false),
+      clearConfigCache: () => Promise.resolve(undefined),
+      formatWithCursor: () => Promise.resolve({ formatted: '', cursorOffset: 0 }),
+      getFileInfo: async () => ({ ignored: false, inferredParser: null }),
+      getSupportInfo: () => Promise.resolve({ languages: [], options: [] }),
+      resolveConfigFile: async () => null,
+    };
+  }
 }
 
 /**
@@ -30,7 +68,7 @@ export async function formatFileContent(filePath: string, content: string): Prom
   }
 }
 
-async function formatWithEditorConfig(filePath: string, content: string) {
+async function formatWithEditorConfig(filePath: string, content: string): Promise<string> {
   const { resolveConfig, format } = await getPrettier();
 
   const config = await resolveConfig(filePath, { editorconfig: true });

